labs/behaviors/constraint-validation_test.ts

```diff
@@ -174,4 +222,26 @@ describe('mixinConstraintValidation()', () => {
         .toBe('Error');
     });
   });
+
+  describe('customError', () => {
+    it('should set customError to true when validator has customError', () => {
+      const control = new TestCustomErrorConstraintValidation();
+      expect(control.validity.customError)
+        .withContext('validity.customError')
+        .toBeTrue();
+    });
+    it('should dispatch invalid event when validator has customError', () => {
+      const control = new TestCustomErrorConstraintValidation();
+      const invalidListener = jasmine.createSpy('invalidListener');
+      control.addEventListener('invalid', invalidListener);
+      control.reportValidity();
+      expect(invalidListener).toHaveBeenCalledWith(jasmine.any(Event));
+    });
+    it('should report custom validation message over other validation messages', () => {
+      const control = new TestCustomErrorConstraintValidation();
```

Comment:
I think this test needs `control.required = true` to ensure other validation is running

Reply:
Thanks @asyncLiz  

I think this is covered in other tests above in the same file. Here we are just making sure that a `customError` is properly reported as an error (the new test validator in the test file just does that: `this.control.setCustomValidity('validator custom error');`).  

Those 3 new tests fail without the fix in labs/behaviors/constraint-validation.ts. They pass after changing it to: 

```js
const customError = !!this[privateCustomValidationMessage] || validity.customError;
```